core/src/main/java/de/eldoria/bloodnight/mobs/MobRegistry.java

```diff
@@ -33,36 +31,22 @@ public MobRegistry() {
      * @throws MobIdAlreadyTakenException when the id is already taken.
      */
     public void add(CustomMob mob) {
-        check(mob);
-        if (mobs.containsKey(mob.id())) throw new MobIdAlreadyTakenException(mob, get(mob.id()));
-        mobs.put(mob.id(), mob);
-    }
-
+        Checks.notNull(mob, "Mob can not be null");
 
-    /**
-     * Returns the {@link CustomMob} associated with the given id.
-     *
-     * @param id the id of the mob to be retrieved.
-     * @return the {@link CustomMob} associated with the given id, or null if no mob is found.
-     */
-    public CustomMob get(String id) {
-        return mobs.get(id);
+        var previousMob = mobs.putIfAbsent(mob.id(), mob);
+        if (previousMob != null) {
+            throw new MobIdAlreadyTakenException(mob, previousMob);
+        }
```

Comment:
I see no enhancement in this change, beside that it is harder to read.

Reply:
Well, that's just the way I write such things, so the map is only queried ones. But I can revert it, if you like.

Reply:
No the map is queried twice internalls xD put if absent just calls get and put afterwards if not value was returned. it is just easier to read if you check if the key is contained, because that is actually what you wanna check.

Reply:
The implementation internally isnt't that easy :D But I revert that change, for those who don't know hashmaps methods beside put, get and contains :P

Reply:
I just checked the default implementation. of course HashMap does more. But I see no real benefit and prefer readability over a probably negligible performance enhancement